design/secrets_decoding_design.md

```diff
@@ -0,0 +1,204 @@
+# Secrets Provider - Base64 Secrets Decoding Design
+
+## Table of Contents
+
+- [Secrets Provider - Base64 Secrets Decoding Design](#secrets-provider---base64-secrets-decoding-design)
+  * [Table of Contents](#table-of-contents)
+  * [Useful links](#useful-links)
+  * [Background](#background)
+  * [Solution](#solution)
+    + [Design](#design)
+    + [Backwards compatibility](#backwards-compatibility)
+  * [Test Plan](#test-plan)
+    + [Annotation Tests](#annotation-tests)
+    + [Decoding Tests](#decoding-tests)
+    + [e2e Tests](#e2e-tests)
+    + [Audit](#audit)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Tasks](#tasks)
+    + [Add a struct for storing content-types of individual Conjur secrets](#add-a-struct-for-storing-content-types-of-individual-conjur-secrets)
+    + [Update K8s Secret mode config to handle pod annotations](#update-k8s-secret-mode-config-to-handle-pod-annotations)
+    + [Add decoding of secrets with 'base64' content-types](#add-decoding-of-secrets-with--base64--content-types)
+    + [Update e2e tests and dev environment scripts/manifests](#update-e2e-tests-and-dev-environment-scripts-manifests)
+    + [Create/update docs](#create-update-docs)
+    + [Docs walkthrough](#docs-walkthrough)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+## Useful links
+
+| Name                                     | Link                                                         |
+| -----------------------------------------| ------------------------------------------------------------ |
+| Secrets Provider Configuration Reference | https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Integrations/k8s-ocp/cjr-k8s-secrets-provider-ref.htm |
+| Feature Functional Spec (internal)       | https://{cyberark_confluence_host}/display/rndp/Secrets+Provider%3A+Secrets+decoding+for+k8s+Functional+Specification |
+
+## Background
+Secrets Provider provides Kubernetes-based applications with access to secrets that are stored and managed in Conjur. These secrets can be any binary/string data. In some cases it may be beneficial, or necessary, to store secret values in Conjur as base64 encoded strings depending on the nature and origin of the secret. Vault also supports multiline and base64 encoded strings which may then be synchronized as Conjur secrets in their encoded state.
```

Comment:
Vault (EPV) doesn't just support base64 encoding, it actually requires that binary secrets be stored this way. So when the synchronizer comes into play, Conjur *needs* to support base64 decoding in order to be compatible with the Vault.

So this feature is less of a flexibility enhancement and more of a necessity for EPV / Synchronizer support.